### DISCUSSION TITLE
[Chapter 6] 이런 경우에도 전략패턴이 좋은 방법일까?

### DISCUSSION THREAD

**Discussion by firewoody237** [cowaine/is-my-code-that-strange#47] [Chapter 06 (조건 분기)] [answered]
최근에 필요에 의해 아래와 같은 코드를 작성한 적이 있습니다.

```java
switch (Logger.Level logLevel) {
    case WARN:
        log.warn("{}", ~~);
        break;
    case ERROR:
        log.error("{}", ~~);
        break;
    ....
}
```

이 코드도 전략패턴을 사용해서 얻는 장점이 크다고 생각하시는지 궁금합니다.
우선 서비스레이어 한곳에서만 사용하는건데, 저 코드를 위해 로그레벨별 클래스를 새로 만든다..?
일단 저는 긍정적이지 않았기 때문에 다른분들의 의견또한 궁금합니다.

**Comment by corock** [comment, marked as answer]
코드 스니펫에 해당하는 컨텍스트는 정확히 파악되지 않지만, 이런 경우는 굳이 전환할 필요 없다고 봅니다.
전략 패턴(_strategy pattern_)은 분기 케이스가 2개보다 더 많거나, 분기된 로직 하나하나가 복잡할 때 효과가 있다고 생각합니다.
협업하는 동료가 패턴을 모를 수도 있고, 기준 없이 무지성 패턴을 쓰게 되면 오히려 오버 엔지니어링이 될 수 있을 것 같아요.
